### PR TITLE
Optimize Table Locator Call, tested with Active and Passive Mode

### PR DESCRIPTION
**Pull request by Indianer3c**

### CODE REVIEW

**Comment by dereuromark** [comment]
Whats the optimization here?

File: src/Controller/Component/CaptchaComponent.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ public function beforeRender(EventInterface $event): void {
 	 */
 	public function addValidation(Validator $validator, ?string $type = null) {
 		/** @var \Captcha\Model\Table\CaptchasTable $Captchas */
-		$Captchas = TableRegistry::getTableLocator()->get('Captcha.Captchas');
+		$Captchas = TableRegistry::getTableLocator()->get('Captchas', ['className' => 'Captcha\Model\Table\CaptchasTable', 'table' => 'captchas']);
```
Comment (dereuromark) [inline src/Controller/Component/CaptchaComponent.php:83]: It is a bit weird to have all 3 the same right now, and then changing the call of one of those 3.

**Comment by Indianer3c** [comment]
> Whats the optimization here?

In Active Mode if the captcha solution is wrong I get the following error message

```
Sth went wrong: Array
(
[captcha_result] => Array
(
[_required] => This field is required
)

)
BadMethodCallException
```
This no longer occurs due to the renewed adjustment.

File: src/Controller/Component/CaptchaComponent.php (hunk 1 above)
Comment (Indianer3c) [reply src/Controller/Component/CaptchaComponent.php:83]: I'm sorry for the confusion. Hope this is the best solution now.

**Comment by dereuromark** [comment]
But how does it optimize it? And why only changing 1/3 of those calls?
I dont quite get the change.

**Comment by Indianer3c** [comment]
> But how does it optimize it? 

When trying out the Captcha Plugin, I initially had problems using the passive mode. Since I myself prefer the passive mode.

When debugging, I often ended up in the Component beforeFilter method. There it seems to be checked whether the behavior is already being used. But since my contact form does not have its own database table, for example, this always failed at the beginning in this line https://github.com/dereuromark/cakephp-captcha/blob/master/src/Controller/Component/CaptchaComponent.php#L44 because no model exists.

> And why only changing 1/3 of those calls?

Ah okay I see there are more calls in the helper and behavior.

> I dont quite get the change.

I'm just getting started again with CakePHP. I can't make a better statement at the moment. You are welcome to get an idea for yourself and make the appropriate adjustments. I myself work first with my fork branch.

**Comment by dereuromark** [comment]
If there is currently no "bug" here, I would prefer using the recommended simple calls.

**Comment by Indianer3c** [comment]
Agreed, I think it's ok. Then all 3 calls are uniform.

**Pull request closed without merging**